ballerina-tests/tests/http_payload_retrieval_after_responding_test.bal

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/lang.runtime as runtime;
+import ballerina/http;
+
+listener http:Listener payloadRetrievalListener = new(payloadRetrievalAfterRespondingTestPort);
+http:Client payloadRetrievalBackendClient = check new("http://localhost:" + payloadRetrievalAfterRespondingTestPort.toString());
+http:Client payloadRetrievalTestClient = check new("http://localhost:" + payloadRetrievalAfterRespondingTestPort.toString());
+
+json? jsonPaylod = ();
+xml? xmlPayload = ();
+string? textPayload = ();
+byte[]? binaryPayload = ();
+
+service /passthrough on payloadRetrievalListener {
+    resource function 'default . () returns string|error {
+        json jsonStr = {a: "a", b: "b"};
+        json jsonPayload = check payloadRetrievalBackendClient->post("/backend/getJson", jsonStr);
+
+        xml xmlStr = xml `<name>Ballerina</name>`;
+        xml xmlPayload = check payloadRetrievalBackendClient->post("/backend/getXml", xmlStr);
+
+        string stringPayload = check payloadRetrievalBackendClient->post("/backend/getString", "want string");
+
+        byte[] binaryPaylod = check payloadRetrievalBackendClient->post("/backend/getByteArray", "BinaryPayload is textVal".toBytes());
+
+        return "Request Processed successfully";
+    }
+}
+
+service /backend on payloadRetrievalListener {
+    resource function 'default getJson(http:Caller caller, http:Request req) returns error? {
+        http:Response response = new;
+        response.setJsonPayload({id: "chamil", values: {a: 2, b: 45, c: {x: "mnb", y: "uio"}}});
+        check caller->respond(response);
+        jsonPaylod = check req.getJsonPayload();
```

Comment:
```suggestion
        requesJsonPaylod = check req.getJsonPayload();
```
Shall we prefix with `request` so the test is more readable

Reply:
Updated all the places.